test/11_array.jl

```diff
@@ -227,6 +227,11 @@ using CategoricalArrays: DefaultRefType, leveltype
     empty!(x)
     @test length(x) == 0
     @test levels(x) == ["e", "a", "b", "c", "zz"]
+    
+    x = categorical([1])
+    @test sizehint!(x, 1000) === x
+    @test_throws MethodError empty!(categorical([1 2; 3 4]))
+    @test_throws MethodError sizehint!(categorical([1 2; 3 4]))
```

Comment:
Can you put the equivalent in 12_missingarray.jl?

Maybe also test that `x == [1]` after the call.